src/assignment/components/AssignmentActions.tsx

```diff
@@ -176,30 +179,36 @@ const AssignmentActions: FunctionComponent<AssignmentActionsProps> = ({
 		/>
 	);
 
-	const renderPublishButton = (config?: Partial<ButtonProps>) => (
-		<Button type="secondary" {...config} />
-	);
+	const renderPublishButton = (config?: Partial<ButtonProps>) => {
+		if (route !== APP_PATH.ASSIGNMENT_DETAIL.route) {
```

Comment:
you're checking if the current page is the detail route
but this component isn't used on the detail page

is this a mistake?